plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -42,8 +43,14 @@
             .withBackoff(1, 5, ChronoUnit.SECONDS)
             .withMaxRetries(5)
             .handleIf(throwable -> getCausalChain(throwable).stream()
-                            .filter(SQLException.class::isInstance)
-                            .anyMatch(exception -> exception.getMessage().contains("Rerun the transaction.")))
+                    .filter(SQLException.class::isInstance)
+                    .anyMatch(exception -> {
+                        String message = nullToEmpty(exception.getMessage());
+                        return message.contains("Rerun the transaction.") ||
```

Comment:
Nit: splitting the `||` across lines seems a bit funny. Maybe make this a constant
```JAVA
// Example: Failed to execute statement: ALTER DATABASE database_xyz SET READ_COMMITTED_SNAPSHOT ON
//  User does not have permission to alter database 'database_xyz', the database does not exist, or the database is not in a state that allows access checks.
private static final String SET_SNAPSHOT_ERROR = "database is not in a state that allows access checks";
```
```java
return message.contains("Rerun the transaction.") || message.contains(SET_SNAPSHOT_ERROR);
```

Reply:
> splitting the || across lines seems a bit funny

thanks for letting me make a joke here. We will surely add more entries and the pattern currently in used allows for best commented & most readable approach with 3+ (arguably, with 2+) entries